services/edit/index.js

```diff
@@ -234,51 +235,85 @@ function createPage() {
 }
 
 /**
- * create a child component (from a base ref) and return it's newly generated ref
- * @param {string} baseRef e.g. [siteprefix]/components/name
- * @returns {Promise}
+ * synchronously determine if a component has references to default components
+ * inside component lists. these child components will be cloned
+ * when creating the parent component
+ * @param {object} data
+ * @returns {object}
  */
-function createChildComponent(baseRef) {
-  return cache.getDataOnly(baseRef).then(function (baseData) {
-    return cache.createThrough(baseRef + '/instances', baseData).then(function (res) {
-      return res._ref;
-    });
+function getChildComponents(data) {
+  let mapping = {};
+
+  _.forOwn(data, function checkProperties(val, key) {
+    // loop through the (base) data in the component
+    if (_.isArray(val)) {
+      // find arrays
+      _.each(val, function checkArrays(item, index) {
```

Comment:
Is there a functional or style reason for using named function on the forOwn and each iterations?